go/adbc/driver/bigquery/driver.go

```diff
@@ -0,0 +1,180 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package bigquery
+
+import (
+	"fmt"
+	"runtime/debug"
+	"strings"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-adbc/go/adbc/driver/internal/driverbase"
+	"github.com/apache/arrow/go/v17/arrow/memory"
+)
+
+const (
+	OptionStringAuthType  = "adbc.bigquery.sql.auth_type"
+	OptionStringProjectID = "adbc.bigquery.sql.project_id"
+	OptionStringDatasetID = "adbc.bigquery.sql.dataset_id"
+	OptionStringTableID   = "adbc.bigquery.sql.table_id"
+
+	OptionValueAuthTypeDefault = "adbc.bigquery.sql.auth_type.auth_bigquery"
```

Comment:
along with adding the bigquery driver itself to the documentation, we should probably add these options to the documentation also. Updating the documentation can be done as a follow-up to this though, it doesn't need to be done in this PR directly.